### PR TITLE
Merge forward 3004.2

### PR DESCRIPTION
**Pull request by dwoz**
### What issues does this PR fix or reference?
Fixes:

### Previous Behavior
Remove this section if not relevant

### New Behavior
Remove this section if not relevant

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes/No

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Review by whytewolf — DISMISSED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,31 @@ Versions are `MAJOR.PATCH`.
 # Changelog
 
 Salt 3004.1 (2022-02-16)
+Salt 3004.2 (2022-05-12)
```
Comment (Ch3LL) [inline CHANGELOG.md:11]: These should be separate entries in the changelog, not under the same title.

File: cicd/jenkins/pr-alma-8-x86_64-py3-pytest [hunk 2]
```diff
@@ -1,6 +1,13 @@
+<<<<<<<< HEAD:cicd/jenkins/pr-alma-8-x86_64-py3-pytest
 @Library('salt@master-1.11') _
 
 runTestSuite(
+========
+@Library('salt@master-1.10') _
+
+runTestSuite(
+    ami_image_id: 'ami-02fc0a57f9b1fa4ed',
+>>>>>>>> 3003.5:cicd/jenkins/pr-centosstream-8-x86_64-py3-pytest
```
Comment (Ch3LL) [inline cicd/jenkins/pr-alma-8-x86_64-py3-pytest:10]: merge conflict lines

**Review by Ch3LL — CHANGES_REQUESTED** [review]

File: tests/pytests/integration/modules/test_virt.py [hunk 3]
```diff
@@ -10,6 +10,10 @@
 
 docker = pytest.importorskip("docker")
 
+pytestmark = pytest.mark.skipif(
+    True, reason="Skipping tests in 3002.x and 3003.x branches."
+)
+
```
Comment (twangboy) [inline tests/pytests/integration/modules/test_virt.py:16]: Doesn't this skip the test moving forward? If we only want to skip in those branches, we should remove this.

File: tests/pytests/scenarios/setup/test_install.py [hunk 4]
```diff
@@ -28,6 +28,8 @@ def use_static_requirements_ids(value):
 
 @pytest.fixture(params=[True, False], ids=use_static_requirements_ids)
 def use_static_requirements(request):
+    if salt.utils.platform.is_windows() and request.param is False:
+        pytest.skip("Skipping test in 3003.x on Windows.")
```
Comment (twangboy) [inline tests/pytests/scenarios/setup/test_install.py:32]: We don't want to skip this test on master, right?

**Review by twangboy — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -427,6 +458,42 @@ Added
   metadata for a package by extracting library requirement information from the
   binary ELF files in the package. (#59569)
 
+- Sign authentication replies to prevent MiTM (cve-2020-22935)
```
Comment (Ch3LL) [inline CHANGELOG.md:461]: this looks to be under the wrong section 3003 which was released a while ago. I think these just need to be removed

File: tests/integration/states/test_file.py [hunk 6]
```diff
@@ -4682,7 +4682,6 @@ def test_file_managed_keep_source_false_http(self):
 
 
 @skipIf(not salt.utils.path.which("patch"), "patch is not installed")
-@pytest.mark.windows_whitelisted
```
Comment (Ch3LL) [inline tests/integration/states/test_file.py:4685]: This needs to be re-added. These tests run on master on windows now

**Review by Ch3LL — CHANGES_REQUESTED** [review]

**Review by Ch3LL — APPROVED** [review]

File: tests/pytests/integration/ssh/test_state.py [hunk 7]
```diff
@@ -121,8 +121,8 @@ def test_state_with_import_from_dir(salt_ssh_cli, nested_state_tree):
     ret = salt_ssh_cli.run(
         "--extra-filerefs=salt://foo/map.jinja", "state.apply", "foo"
     )
-    assert ret.returncode == 0
-    assert ret.data
+    assert ret.exitcode == 0
+    assert ret.json
```
Comment (s0undt3ch) [inline tests/pytests/integration/ssh/test_state.py:125]: These shouldn't have been changed, but it's not a blocker

File: tests/pytests/unit/transport/test_tcp.py [hunk 8]
```diff
@@ -335,3 +374,58 @@ def connect(*args, **kwargs):
             client.io_loop.run_sync(client.connect)
     finally:
         client.close()
+
+
+async def test_when_async_req_channel_with_syndic_role_should_use_syndic_master_pub_file_to_verify_master_sig(
```
Comment (s0undt3ch) [inline tests/pytests/unit/transport/test_tcp.py:379]: These long names.....

**Review by s0undt3ch — APPROVED** [review]
Left comments but they are not blockers.

**Review by twangboy — APPROVED** [review]

File: tests/pytests/unit/transport/test_tcp.py (hunk 8 above)
Comment (dwoz) [reply tests/pytests/unit/transport/test_tcp.py:379]: Yeah @s0undt3ch, we need to have a talk with @garethgreenaway about naming conventions ;)

Comment (s0undt3ch) [reply tests/pytests/unit/transport/test_tcp.py:379]: And @waynew.

**Comment by max-arnold** [comment]
FYI, this change adds the following ERROR message to each salt-call run:

```
salt-call test.ping
[ERROR   ] WTF False <bound method AsyncAuth.authenticate of <salt.crypt.AsyncAuth object at 0x7f7cc1bf0dc0>>
local:
    True
```

```
salt-call --versions-report
Salt Version:
          Salt: 3005+0na.264d02a

Dependency Versions:
          cffi: Not Installed
      cherrypy: Not Installed
      dateutil: Not Installed
     docker-py: Not Installed
         gitdb: Not Installed
     gitpython: Not Installed
        Jinja2: 3.1.2
       libgit2: Not Installed
      M2Crypto: Not Installed
          Mako: Not Installed
       msgpack: 1.0.4
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     pycparser: Not Installed
      pycrypto: Not Installed
  pycryptodome: 3.15.0
        pygit2: Not Installed
        Python: 3.8.10 (default, Mar 15 2022, 12:22:08)
  python-gnupg: Not Installed
        PyYAML: 6.0
         PyZMQ: 23.2.0
         smmap: Not Installed
       timelib: Not Installed
       Tornado: 4.5.3
           ZMQ: 4.3.4

System Versions:
          dist: ubuntu 20.04 Focal Fossa
        locale: utf-8
       machine: x86_64
       release: 5.4.0-77-generic
        system: Linux
       version: Ubuntu 20.04 Focal Fossa
```

**Pull request merged**